frontend/src/components/App.svelte

```diff
@@ -0,0 +1,83 @@
+<script>
+	import { fade } from 'svelte/transition';
+
+	import Name from './Name';
+	import Header from './Header';
+	import Rooms from './Rooms';
+	import Footer from './Footer';
+
+	const roomsFadeDelay = 200;
+	let submitted = false;
```

Comment:
Initialize this depending on whether username is already set. Otherwise the name input form is always shown even when the name is already set.

```suggestion
	let submitted = !!userState.name;
```

Reply:
Is !! really the inverse operator in js?

Reply:
Nope, the inverse is still `!`. But writing `!!` is a fancy way of doing a bool cast :)

Reply:
```
!"foo"  === false 
!!"foo" === true

!null === true
!!null === false
```

Reply:
Genius!